projects/transformers/experiments/sparse_bert.py

```diff
@@ -48,8 +48,34 @@
 )
 
 
+static_sparse_encoder_bert_100k = deepcopy(bert_100k)
+static_sparse_encoder_bert_100k.update(
+    # Model Arguments
+    overwrite_output_dir=False,
+    model_type="static_sparse_encoder_bert",
+    config_kwargs=dict(
+        sparsity=0.8,
+    ),
+    trainer_callbacks=[RezeroWeightsCallback()]
+)
+
+# Sparse Bert of only two layers and one attention head.
+mini_sparse_bert_debug = deepcopy(debug_bert)
+mini_sparse_bert_debug.update(
+    model_type="static_sparse_encoder_bert",
+    config_kwargs=dict(
```

Comment:
Not changing the hidden size of intermediate and output layers? The ratio of number of weights in attention layers to number of weights in non-attention layers will be significantly unbalanced if you don't change those as well (see the calculations we did yesterday).

Reply:
This was just a debug model to make sure things can run. Do you think adjusting those layers is needed?

Reply:
If is a debug thing, won't matter (except again for speed). But it won't hurt to add two extra parameters and can avoid misleading someone new to the project reading the code and looking for an example of a mini sparse bert.

Reply:
Fair enough. Still a bit shaky in my understanding in how the number of attention heads relates to the hidden size. Can you explain again?

Reply:
If you change num_attention_heads and look the num of params, they are the same. I think that is just defining in how many heads the weights are divided by. Would have to go into the model code to confirm this hypothesis. 

So you have to change the hidden_size to actually change the number of params. Then if you change the hidden size without changing the intermediate size, you will have a disproportionate amount of weights in the linear transformations post attention compared to attention. 

I didn't spend time looking at the model, I am assuming you would that.

Reply:
I see. That helps. Particularly, "If you change num_attention_heads and look the num of params, they are the same". I double-checked, you're right about this. I'll update the mini model.